core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -38,7 +38,13 @@ object ConsoleConsumer extends Logging {
 
   def main(args: Array[String]) {
     val conf = new ConsumerConfig(args)
-    run(conf)
+    try {
+      run(conf)
+    } catch {
+      case e: Throwable =>
+        error("Error running consumer: ", e)
+    }
+    System.exit(0);
```

Comment:
Having looked, I don't think the exception case is specified one way or another. What is specified is that braces should be skipped for single-expression if/else blocks. The reason why I think it should be removed is that it adds noise and it doesn't help in any way (if you add a new expression without adding the braces, there will be a compiler error due to the catch being in an unexpected place, so no safety problems). The reason why I think the existing code is the way you say is probably that many people are more familiar with Java than Scala. The one example you mention is actually quite recent and it was my suggestion. :) Anyway, it's up to you, since this is an area that is not well specified.

`System.exit(1)` is fine. I prefer `-1` as it's clearer to me that it's an error, but as you say it looks like Kafka code tends to use `1`.